contracts/src/v1/Handlers.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+pragma solidity 0.8.25;
+
+import {IERC20} from "../interfaces/IERC20.sol";
+
+import {SafeTokenTransferFrom} from "../utils/SafeTransfer.sol";
+
+import {AssetsStorage, TokenInfo} from "../storage/AssetsStorage.sol";
+import {CoreStorage} from "../storage/CoreStorage.sol";
+import {PricingStorage} from "../storage/PricingStorage.sol";
+import {SubstrateTypes} from "../SubstrateTypes.sol";
+import {MultiAddress} from "../types/Common.sol";
+import {Address} from "../utils/Address.sol";
+import {AgentExecutor} from "../AgentExecutor.sol";
+import {Agent} from "../Agent.sol";
+import {Call} from "../utils/Call.sol";
+import {Token} from "../Token.sol";
+import {Upgrade} from "../Upgrade.sol";
+import {Functions} from "../Functions.sol";
+import {IGatewayBase} from "../interfaces/IGatewayBase.sol";
+import {IGatewayV1} from "./IGateway.sol";
+
+import {
+    ParaID,
+    Ticket,
+    Costs,
+    AgentExecuteCommand,
+    AgentExecuteParams,
+    CreateAgentParams,
+    CreateChannelParams,
+    UpdateChannelParams,
+    UpgradeParams,
+    SetOperatingModeParams,
+    TransferNativeFromAgentParams,
+    SetTokenTransferFeesParams,
+    SetPricingParametersParams,
+    UnlockNativeTokenParams,
+    RegisterForeignTokenParams,
+    MintForeignTokenParams
+} from "./Types.sol";
+
+library HandlersV1 {
+    using Address for address;
+    using SafeTokenTransferFrom for IERC20;
+
+    function agentExecute(address executor, bytes calldata data) external {
+        AgentExecuteParams memory params = abi.decode(data, (AgentExecuteParams));
+
+        address agent = Functions.ensureAgent(params.agentID);
+
+        if (params.payload.length == 0) {
+            revert IGatewayBase.InvalidAgentExecutionPayload();
+        }
+
+        (AgentExecuteCommand command, bytes memory commandParams) =
+            abi.decode(params.payload, (AgentExecuteCommand, bytes));
+        if (command == AgentExecuteCommand.TransferToken) {
+            (address token, address recipient, uint128 amount) =
+                abi.decode(commandParams, (address, address, uint128));
+            Functions.withdrawNativeToken(executor, agent, token, recipient, amount);
+        }
+    }
+
+    /// @dev Create an agent for a consensus system on Polkadot
+    function createAgent(bytes calldata data) external {
+        CreateAgentParams memory params = abi.decode(data, (CreateAgentParams));
+        Functions.createAgent(params.agentID);
+    }
+
+    /// @dev Perform an upgrade of the gateway
+    function upgrade(bytes calldata data) external {
+        UpgradeParams memory params = abi.decode(data, (UpgradeParams));
+        Upgrade.upgrade(params.impl, params.implCodeHash, params.initParams);
+    }
+
+    // @dev Set the operating mode of the gateway
+    function setOperatingMode(bytes calldata data) external {
```

Comment:
Both v1 and v2 paths modify the same storage. So essentially v1 and v2 handlers are just two ways to set the bridge operating mode, and they are not ways to disable v1 and v2 separately.